src/playercards/cards/CharlieKane.ttslua

```diff
@@ -0,0 +1,299 @@
+    children   = {{
+    tag = "Button",
+    attributes = {
+      onClick  = "stopListening",
+      font     = "font_teutonic-arkham",
+      fontSize = 150,
+      text = "End Skill Test"
+    }
+  }},
+    attributes = {
+      id              = "Helper2",
+      active          = false,
+      rotation        = "0 0 90",
+      scale           = "0.175 0.175 1",
+      childAlignment  = "MiddleCenter",
+      position        = "90 -70 -40",
+      width           = 350 * 2 + 55,
+      height          = 185,
+      spacing         = "55"
+    }
+  }
+
+  local currentXml = self.UI.getXmlTable()
+  table.insert(currentXml, buttonXml)
+  self.UI.setXmlTable(currentXml)
+end
+
+-- starts listening for Ally card exhausts
+function extraClickFunction()
+  addExhaustButtons()
+  bigTable = {}
+  matchingSkillTotal = 0
+  if hasButton then
+    self.removeButton(0)
+  end
+  if self.UI.getAttribute("Helper2", "active") == "false" then
+    self.UI.setAttribute("Helper2", "active", "true")
+  end
+  isListening = true
+  self.createButton({
+    function_owner = self,
+    label          = "+0",
+    click_function = "none",
+    position       = Vector(0, 1, .75),
+    height         = 0,
+    width          = 0,
+    font_size      = 550,
+    font_color     = "Black",
+    rotation       = {0, 90, 0}
+  })
+  hasButton = true
+end
+
+function addExhaustButtons()
+  local matColor = PlayermatApi.getMatColorByPosition(self.getPosition())
+  for _, object in ipairs(PlayermatApi.searchAroundPlayermat(matColor)) do
+    if object.type == "Card" then
+      local md = JSON.decode(object.getGMNotes()) or {}
+      if string.match(md.traits, "Ally") then
+        if not hasSpecificXmlElement(object, "Charlie") then
+          local xml = object.UI.getXmlTable() or {}
+          local exhaustButton =
+            {
+              tag = "VerticalLayout",
+              attributes = {
+                height = 375,
+                width = 375,
+                padding = "0 0 20 25",
+                scale = ".35 .35 1",
+                rotation = "0 0 180",
+                position = "0 20 -100",
+                color = "rgba(0,0,0,0.9)",
+                onClick = self.getGUID() .. "/exhaustOrReadyAlly(" .. object.getGUID() .. ")"
+              },
+              children = {
+                {
+                  tag = "Text",
+                  attributes = {
+                    fontSize = "135",
+                    font = "font_arkhamicons",
+                    color = "#ffffff",
+                    text = "v"
+                  }
+                },
+                {
+                  tag = "Text",
+                  attributes = {
+                    id = "charlieText",
+                    fontSize = "125",
+                    font = "font_teutonic-arkham",
+                    color = "#ffffff",
+                    text = "Exhaust"
+                  }
+                }
+              }
+            }
```

Comment:
Probably turn this into a separate function for readability. (needs the GUID as parameter and returns the xml table for the button)